cloudbuild.yaml

```diff
@@ -0,0 +1,111 @@
+# Run tests from the repo, based on the branch name
+- name: 'gcr.io/google.com/hadoop-cloud-dev/init-actions-image'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |
+    [[ "$BRANCH_NAME" == .*bigtable.* ]] && python bigtable/test_bigtable.py || exit 0
+- name: 'gcr.io/google.com/hadoop-cloud-dev/init-actions-image'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |
+    [[ "$BRANCH_NAME" == .*drill.* ]] && python drill/test_drill.py || exit 0
+- name: 'gcr.io/google.com/hadoop-cloud-dev/init-actions-image'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |
+    [[ "$BRANCH_NAME" == .*flink.* ]] && python flink/test_flink.py || exit 0
+- name: 'gcr.io/google.com/hadoop-cloud-dev/init-actions-image'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |
+    [[ "$BRANCH_NAME" == .*ganglia.* ]] && python ganglia/test_ganglia.py || exit 0
+- name: 'gcr.io/google.com/hadoop-cloud-dev/init-actions-image'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |
+    [[ "$BRANCH_NAME" == .*gpu.* ]] && python gpu/test_install_gpu_driver.py || exit 0
+- name: 'gcr.io/google.com/hadoop-cloud-dev/init-actions-image'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |
+    [[ "$BRANCH_NAME" == .*hbase.* ]] && python hbase/test_hbase.py || exit 0
+- name: 'gcr.io/google.com/hadoop-cloud-dev/init-actions-image'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |
+    [[ "$BRANCH_NAME" == .*drill.* ]] && python hive-hcatalog/test_hive.py || exit 0
+- name: 'gcr.io/google.com/hadoop-cloud-dev/init-actions-image'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |
+    [[ "$BRANCH_NAME" == .*hue.* ]] && python hue/test_hue.py || exit 0
+- name: 'gcr.io/google.com/hadoop-cloud-dev/init-actions-image'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |
+    [[ "$BRANCH_NAME" == .*kafka.* ]] && python kafka/test_kafka.py || exit 0
+- name: 'gcr.io/google.com/hadoop-cloud-dev/init-actions-image'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |
+    [[ "$BRANCH_NAME" == .*livy.* ]] && python livy/test_livy.py || exit 0
+- name: 'gcr.io/google.com/hadoop-cloud-dev/init-actions-image'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |
+    [[ "$BRANCH_NAME" == .*oozie.* ]] && python oozie/test_oozie.py || exit 0
+- name: 'gcr.io/google.com/hadoop-cloud-dev/init-actions-image'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |
+    [[ "$BRANCH_NAME" == .*presto.* ]] && python presto/test_presto.py || exit 0
+- name: 'gcr.io/google.com/hadoop-cloud-dev/init-actions-image'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |
+    [[ "$BRANCH_NAME" == .*ranger.* ]] && python ranger/test_ranger.py || exit 0
+- name: 'gcr.io/google.com/hadoop-cloud-dev/init-actions-image'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |
+    [[ "$BRANCH_NAME" == .*solr.* ]] && python solr/test_solr.py || exit 0
+- name: 'gcr.io/google.com/hadoop-cloud-dev/init-actions-image'
+  entrypoint: 'bash'
+  args:
+  - '-c'
+  - |
+    [[ "$BRANCH_NAME" == .*starburst.* ]] && python starburst-presto/test_presto.py || exit 0
```

Comment:
Are you intentionally ignoring the exit code from python?

Reply:
I was previously making sure that the steps were all executed in case one of the tests failed. I now changed the step to only fail if the python test fails. If the python test is simply not run if the branch name doesn't match, the step still succeeds.

Reply:
The 'waitFor' attribute defines the dependencies of each step. In this case, the tests run after the container is built.